upb/table.c

```diff
@@ -506,10 +517,35 @@ bool upb_strtable_insert(upb_strtable *t, const char *k, size_t len,
   if (tabkey == 0) return false;
 
   hash = table_hash(key.str.str, key.str.len);
-  insert(&t->t, key, tabkey, v, hash, &strhash, &streql);
+  insert(&t->t, key, tabkey, v, hash, &strhash, &streql, false);
   return true;
 }
 
+upb_tryinsert_status upb_strtable_tryinsert(upb_strtable *t, const char *k,
+                                            upb_value val, upb_arena *a) {
+  lookupkey_t key;
+  upb_tabkey tabkey;
+  uint32_t hash;
+  size_t len = strlen(k);
+
+  if (isfull(&t->t)) {
```

Comment:
this will grow the table even if the key is already present which seems undesirable